### PR TITLE
Add SpaceAroundKeyword cop

### PR DESCRIPTION
**Pull request by lumeet**
It started off as `SpaceAroundDo` but expanded to cover most keywords. Now I've to submit it before it gets even bigger. ;)

I think the cop could basically render `Style/SpaceBeforeModifierKeyword` and `Style/SpaceAfterControlKeyword` useless. Another strategy would be splitting this one into more specific cops. I don't know if there's any need for such, though.

### CODE REVIEW

File: lib/rubocop/cop/style/space_around_keyword.rb [hunk 1]
```diff
@@ -0,0 +1,192 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Checks the spacing around the keywords.
+      #
+      # @example
```
Comment (bbatsov) [inline lib/rubocop/cop/style/space_around_keyword.rb]: I'd add a few more examples here - e.g. `while` and some modifier, so it's clear this handles all sorts of spacing around keywords.

File: lib/rubocop/cop/style/space_around_keyword.rb [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+      #
+      #   # bad
+      #   something 'test'do|x|
+      #   end
+      #
+      #   # good
+      #   something 'test' do |x|
+      #   end
+      class SpaceAroundKeyword < Cop
+        MSG_BEFORE = 'Space before `%s` missing.'.freeze
```
Comment (bbatsov) [inline lib/rubocop/cop/style/space_around_keyword.rb]: Let's change this to "Space before keyword `%s` is missing.".

File: lib/rubocop/cop/style/space_around_keyword.rb [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        MSG_AFTER = 'Space after `%s` missing.'.freeze
```
Comment (bbatsov) [inline lib/rubocop/cop/style/space_around_keyword.rb]: Same here.

**Comment by lumeet** [comment]
Requested changes made and rebased.

**Comment by bbatsov** [comment]
:+1:

**Pull request merged**